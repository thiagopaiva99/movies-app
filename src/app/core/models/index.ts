export * from './movie';
export * from './pagination';
export * from './configuration';
