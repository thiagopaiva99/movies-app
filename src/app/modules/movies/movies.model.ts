export interface Category {
  apiValue: string;
  text: string;
}
