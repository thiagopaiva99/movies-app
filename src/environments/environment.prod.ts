export const environment = {
  production: true,
  apiToken: '977b9d58f71501783437e79b3ed26069',
  apiUrl: 'https://api.themoviedb.org/3',
};
