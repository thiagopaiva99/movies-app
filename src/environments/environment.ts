// This file can be replaced during build by using the `fileReplacements` array.
// `ng build` replaces `environment.ts` with `environment.prod.ts`.
// The list of file replacements can be found in `angular.json`.

export const environment = {
  production: false,
  apiToken: 'eyJhbGciOiJIUzI1NiJ9.eyJhdWQiOiI4OGJlZDIxYjc0MDQ1NGRhNjUwMGQzYWM5NDM0MmE2YyIsInN1YiI6IjVjZTliNTBjOTI1MTQxNTc4YmI5ZWY5MSIsInNjb3BlcyI6WyJhcGlfcmVhZCJdLCJ2ZXJzaW9uIjoxfQ.Vg7IWY9aucXvxnqfyFxeZd8J49pBdsZ16PWyB1toD1Y'
};

/*
 * For easier debugging in development mode, you can import the following file
 * to ignore zone related error stack frames such as `zone.run`, `zoneDelegate.invokeTask`.
 *
 * This import should be commented out in production mode because it will have a negative impact
 * on performance if an error is thrown.
 */
// import 'zone.js/plugins/zone-error';  // Included with Angular CLI.
