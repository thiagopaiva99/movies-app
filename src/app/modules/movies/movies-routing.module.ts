import { NgModule } from '@angular/core';
import { RouterModule, Routes } from '@angular/router';
import { MoviesListComponent } from './containers/movies-list/movies-list.component';

const routes: Routes = [
  {
    path: ':category',
    component: MoviesListComponent
  }
];

@NgModule({
  imports: [RouterModule.forChild(routes)],
  exports: [RouterModule]
})
export class MoviesRoutingModule { }
