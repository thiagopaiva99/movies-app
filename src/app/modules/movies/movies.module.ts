import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';

import { MoviesRoutingModule } from './movies-routing.module';
import { MoviesListComponent } from './containers/movies-list/movies-list.component';
import { MovieDetailsComponent } from './containers/movie-details/movie-details.component';
import { TabPillComponent } from './components/tab-pill/tab-pill.component';


@NgModule({
  declarations: [
    MoviesListComponent,
    MovieDetailsComponent,
    TabPillComponent
  ],
  imports: [
    CommonModule,
    MoviesRoutingModule
  ]
})
export class MoviesModule { }
