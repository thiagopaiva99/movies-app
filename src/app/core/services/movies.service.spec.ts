import { HttpClientTestingModule, HttpTestingController } from '@angular/common/http/testing';
import { TestBed } from '@angular/core/testing';
import { datatype, internet, lorem, name, system } from 'faker';
import { environment } from '../../../environments/environment';

import { MoviesService } from './movies.service';

const mockMoviesFactory = () => ({
  page: 1,
  results: [
    {
      id: 123123,
      poster_path: '/some-file-path',
      backdrop_path: '/some-file-path',
      release_date: '2021-12-17',
      genres: [],
      original_language: 'pt',
      overview: 'some description',
      title: 'some title',
      vote_average: 78,
    }
  ]
})

describe('MoviesService', () => {
  let service: MoviesService;
  let httpMock: HttpTestingController;

  beforeEach(() => {
    TestBed.configureTestingModule({
      imports: [HttpClientTestingModule]
    });

    service = TestBed.inject(MoviesService);
    httpMock = TestBed.inject(HttpTestingController);
  });

  it('should be created', () => {
    expect(service).toBeTruthy();
  });

  it('should get movies by category', (done) => {
    const category = lorem.word();

    service.getMoviesByCategory(category)
    .subscribe((movies) => {
      expect(movies).toMatchSnapshot('Movies List')
      done()
    })

    const httpRequest = httpMock.expectOne(`${environment.apiUrl}/movie/${category}?page=1`)
    httpRequest.flush(mockMoviesFactory())
  })

  it('should get movies by recommendation', (done) => {
    const movieId = datatype.number();

    service.getRecommendations(movieId)
    .subscribe((movies) => {
      expect(movies).toMatchSnapshot('Recommendations List')
      done()
    })

    const httpRequest = httpMock.expectOne(`${environment.apiUrl}/movie/${movieId}/recommendations`)
    httpRequest.flush(mockMoviesFactory())
  })

  it('should get movie by id', (done) => {
    const movieId = datatype.number();

    service.getById(movieId)
    .subscribe((movies) => {
      expect(movies).toMatchSnapshot('Movie By Id')
      done()
    })

    const httpRequest = httpMock.expectOne(`${environment.apiUrl}/movie/${movieId}`)
    httpRequest.flush(mockMoviesFactory())
  })

  it('should search movie by query', (done) => {
    const query = lorem.word()

    service.search(query)
    .subscribe((movies) => {
      expect(movies).toMatchSnapshot('Search by Query')
      done()
    })

    const httpRequest = httpMock.expectOne(`${environment.apiUrl}/search/movie?query=${query}&page=1`)
    httpRequest.flush(mockMoviesFactory())
  })

  it('should get movie credits', (done) => {
    const mock = {
      cast: [
        {
          id: 123213,
          name: 'some name',
          character: 'some character',
          profile_path: '/some-file-path',
        }
      ]
    }

    const movieId = datatype.number();

    service.getCredits(movieId)
    .subscribe((movies) => {
      expect(movies).toMatchSnapshot('Movie Casting')
      done()
    })

    const httpRequest = httpMock.expectOne(`${environment.apiUrl}/movie/${movieId}/credits`)
    httpRequest.flush(mock)
  })
});
