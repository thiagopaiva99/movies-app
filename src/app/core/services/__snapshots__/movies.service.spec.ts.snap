// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MoviesService should get movie by id: Movie By Id 1`] = `
Object {
  "backdropUrl": "undefinedw300undefined",
  "imageUrl": "undefinedw342undefined",
  "page": 1,
  "results": Array [
    Object {
      "backdrop_path": "/some-file-path",
      "genres": Array [],
      "id": 123123,
      "original_language": "pt",
      "overview": "some description",
      "poster_path": "/some-file-path",
      "release_date": "2021-12-17",
      "title": "some title",
      "vote_average": 78,
    },
  ],
}
`;

exports[`MoviesService should get movie credits: Movie Casting 1`] = `
Array [
  Object {
    "character": "some character",
    "id": 123213,
    "imageUrl": "undefinedw342/some-file-path",
    "name": "some name",
    "profilePath": "/some-file-path",
  },
]
`;

exports[`MoviesService should get movies by category: Movies List 1`] = `
Array [
  Object {
    "backdropPath": "/some-file-path",
    "genres": Array [],
    "id": 123123,
    "imageUrl": "undefinedw342/some-file-path",
    "originalLanguage": "pt",
    "overview": "some description",
    "posterPath": "/some-file-path",
    "releaseDate": "2021-12-17",
    "title": "some title",
    "voteAverage": 78,
  },
]
`;

exports[`MoviesService should get movies by recommendation: Recommendations List 1`] = `
Array [
  Object {
    "backdropPath": "/some-file-path",
    "genres": Array [],
    "id": 123123,
    "imageUrl": "undefinedw342/some-file-path",
    "originalLanguage": "pt",
    "overview": "some description",
    "posterPath": "/some-file-path",
    "releaseDate": "2021-12-17",
    "title": "some title",
    "voteAverage": 78,
  },
]
`;

exports[`MoviesService should search movie by query: Search by Query 1`] = `
Array [
  Object {
    "backdropPath": "/some-file-path",
    "genres": Array [],
    "id": 123123,
    "imageUrl": "undefinedw342/some-file-path",
    "originalLanguage": "pt",
    "overview": "some description",
    "posterPath": "/some-file-path",
    "releaseDate": "2021-12-17",
    "title": "some title",
    "voteAverage": 78,
  },
]
`;
